src/vm/types/serialization.rs

```diff
@@ -481,12 +538,29 @@ impl Value {
             Optional(OptionalData{ data: Some(value) }) => {
                 value.serialize_write(w)?;
             },
-            List(data) => {
+            Sequence(List(data)) => {
                 w.write_all(&data.len().to_be_bytes())?;
                 for item in data.data.iter() {
                     item.serialize_write(w)?;
                 }
             },
+            Sequence(Buffer(value)) => {
+                w.write_all(&(u32::from(value.len()).to_be_bytes()))?;
+                w.write_all(&value.data)?
+            },
+            Sequence(SequenceData::String(UTF8(value))) => {
+                let total_len = value.data.iter()
+                    .flatten()
```

Comment:
Does `.flatten()` do an internal copy?  If so, then can we avoid using it?

Reply:
Addressed with